applications/ShapeOptimizationApplication/python_scripts/algorithm_bead_optimization.py

```diff
@@ -112,6 +118,15 @@ def InitializeOptimizationLoop(self):
         self.mapper = mapper_factory.CreateMapper(self.design_surface, self.design_surface, self.mapper_settings)
         self.mapper.Initialize()
 
+        if  self.filter_penalty_term:
```

Comment:
I suggest checking against None explicitly

This will fail for:
self.filter_penalty_term = False

Reply:
this is a bool, why should i check for None?

Reply:
I was looking at another variable 
My bad, sorry for the noise :)

Reply:
I was looking at another variable 
My bad, sorry for the noise :)

Reply:
I was looking at another variable 
My bad, sorry for the noise :)